### PR TITLE
EM-1966: Add checks for comment popup on far right of page

### PR DESCRIPTION
**Pull request by tomxelliott**
- add test cases to cover new constraints on pop-up location

**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: projects/media-viewer/src/lib/annotations/annotation-set/annotation/popup-toolbar/popup-toolbar.component.ts [hunk 1]
```diff
@@ -48,32 +48,56 @@ export class PopupToolbarComponent {
       return top <= 0 ? this.defaultHeight : top;
 
     } else if (this.rotate === 90) {
-      const top = this.rectangle.y * this.zoom + this.rectangle.height/2 * this.zoom + this.defaultWidth/2;
-      return top >= this.height ? this.height : top;
+      const top = this.rectangle.y * this.zoom + this.rectangle.height / 2 * this.zoom + this.defaultWidth / 2;
+      if (top >= this.height) {
+        return this.height;
+      } else if (top < 351.5) {
```
Comment (velimd) [inline projects/media-viewer/src/lib/annotations/annotation-set/annotation/popup-toolbar/popup-toolbar.component.ts]: what are these magic numbers?

File: projects/media-viewer/src/lib/annotations/annotation-set/annotation/popup-toolbar/popup-toolbar.component.ts [hunk 2]
```diff
@@ -48,32 +48,56 @@ export class PopupToolbarComponent {
...
   }
 
   get left() {
     if (this.rotate === 0) {
-      const left = (this.rectangle.x + (this.rectangle.width/2)) * this.zoom - (this.defaultWidth/2);
-      return left <= 0 ? 0 : left;
+      const left = (this.rectangle.x + (this.rectangle.width / 2)) * this.zoom - (this.defaultWidth / 2);
+      if (left <= 0) {
+        return 0;
+      } else if (left > 464) {
```
Comment (velimd) [inline projects/media-viewer/src/lib/annotations/annotation-set/annotation/popup-toolbar/popup-toolbar.component.ts]: same

File: projects/media-viewer/src/lib/annotations/annotation-set/annotation/popup-toolbar/popup-toolbar.component.ts [hunk 3]
```diff
@@ -48,32 +48,56 @@ export class PopupToolbarComponent {
...
       const top = (this.rectangle.y + this.rectangle.height) * this.zoom + 10;
       return top >= this.height - this.defaultHeight ? this.height - this.defaultHeight : top;
 
     } else if (this.rotate === 270) {
-      const top = (this.rectangle.y + (this.rectangle.height/2)) * this.zoom - (this.defaultWidth/2);
-      return top <= 0 ? 0 : top;
+      const top = (this.rectangle.y + (this.rectangle.height / 2)) * this.zoom - (this.defaultWidth / 2);
+      if (top <= 0) {
+        return 0;
+      } else if (top > 705) {
```
Comment (benouaer) [inline projects/media-viewer/src/lib/annotations/annotation-set/annotation/popup-toolbar/popup-toolbar.component.ts]: can we use constants for these values

File: projects/media-viewer/src/lib/annotations/annotation-set/annotation/popup-toolbar/popup-toolbar.component.ts [hunk 4]
```diff
@@ -48,32 +48,56 @@ export class PopupToolbarComponent {
...
+      }
 
     }
   }
 
   get left() {
     if (this.rotate === 0) {
-      const left = (this.rectangle.x + (this.rectangle.width/2)) * this.zoom - (this.defaultWidth/2);
-      return left <= 0 ? 0 : left;
+      const left = (this.rectangle.x + (this.rectangle.width / 2)) * this.zoom - (this.defaultWidth / 2);
```
Comment (benouaer) [inline projects/media-viewer/src/lib/annotations/annotation-set/annotation/popup-toolbar/popup-toolbar.component.ts]: looks like 0 and 180 are exactly the same

**Pull request merged**